python/example_code/lookoutvision/inference.py

```diff
@@ -95,50 +83,204 @@ def download_from_s3(s3_resource, photo):
+
+        if not reject:
+            logger.info("No anomalies found.")
+        return reject
+
+    @staticmethod
+    def reject_on_anomaly_types(image, prediction, confidence_limit, anomaly_types_limit):
+        """
+        Checks if the number of anomaly types > than the anomaly types limit and if
+        the prediction confidence > than the confidence limit.
+        :param image: The name of the image file that was analyzed.
+        :param prediction: The DetectAnomalyResult object returned from DetectAnomalies
+        :param confidence: The minimum acceptable confidence. Float value between 0 and 1.
+        :param anomaly_types_limit: The maximum number of allowable anomaly types (Integer).
+        :return: True if the error condition indicates an anomaly, otherwise False.
+        """
+
+        logger.info("Checking number of anomaly types for %s",image)
+
+        anomaly_types = set()
+        reject = False
+
+        if prediction['IsAnomalous'] and prediction['Confidence'] >= confidence_limit:
+            for anomaly in prediction['Anomalies']:
+                anomaly_types.add(anomaly['Name'])
+
+            # Reduce by one to account for 'background' anomaly type.
+            if (len(anomaly_types)-1) > anomaly_types_limit:
+                reject = True
+                reject_info = (f"Rejected: Anomaly confidence ({prediction['Confidence']:.2%}) "
+                f"is greater than limit ({confidence_limit:.2%}) and "
+                f"the number of anomaly types ({len(anomaly_types)-1}) is "
+                f"greater than the limit ({anomaly_types_limit})")
+
+                logger.info("%s", reject_info)
+
+        if not reject:
+            logger.info("No anomalies found.")
+        return reject
+
+    @staticmethod
+    def reject_on_coverage(image, prediction, confidence_limit, anomaly_label, coverage_limit):
+        """
+        Checks if the coverage area of an anomaly is > than the coverage limit and if
+        the prediction confidence > than the confidence limit.
+        :param image: The name of the image file that was analyzed.
+        :param prediction: The DetectAnomalyResult object returned from DetectAnomalies
+        :param confidence_limit: The minimum acceptable confidence. Float value between 0 and 1.
+        :anomaly_label: The anomaly label for the type of anomaly that you want to check.
+        :coverage_limit: The maximum acceptable percentage coverage of an anomaly (Float 0-1).
+        :return: True if the error condition indicates an anomaly, otherwise False.
+        """
+
+        reject = False
+
+        logger.info("Checking coverage for %s", image)
+
+        if prediction['IsAnomalous'] and prediction['Confidence'] >= confidence_limit:
+            for anomaly in prediction['Anomalies']:
+                if anomaly['Name'] == anomaly_label and \
+                        anomaly['PixelAnomaly']['TotalPercentageArea'] > (coverage_limit/100):
```

Comment:
(optional) Preference for parentheses instead of line continuation, so:
```
                if (anomaly['Name'] == anomaly_label and
                        anomaly['PixelAnomaly']['TotalPercentageArea'] > (coverage_limit/100)):
```

Reply:
Fixed.